### PR TITLE
Add a check to the rest service for http or https

### PR DESCRIPTION
**Pull request by JTF4**
When I was working on a module, I accidentally forgot to add http:// to the IP before sending it to the rest service. I thought that it might be nice to have some checks in place to check if incoming url has `http://` or `https://` at the beginning, and if it doesn't, then append a default `http://` to it.

### CODE REVIEW

**Comment by Julusian** [comment]
Should this be targetting 2.2 or 2.3?
In 2.3 this class is being marked as deprecated with the intention of being removed or rewritten in 2.4

**Comment by krocheck** [comment]
I don't mind bolstering this a little even if its deprecated but I'm not a fan of assuming `http://` when none is provided.  I'd rather invoke the `cb` and send an error back to the module.

**Comment by JTF4** [comment]
I like the CB idea. I'll make it cb an error message and also emit one to debug.

**Comment by JTF4** [comment]
Just changed it to respond with an error

File: lib/Service/Rest.js [hunk 1]
```diff
@@ -9,6 +9,12 @@ class ServiceRest extends CoreBase {
 		this.system.on('rest_get', (url, cb, extra_headers, extra_args) => {
 			this.debug('making request:', url)
 
+
+			if(!url.startsWith('http://') || !url.startsWith('https://')){
+				cb(true, { error: 'Url does not start with http:// or https://' })
```
Comment (krocheck) [inline lib/Service/Rest.js]: @Julusian should this technically be: 
`cb(true, { error: new Error('Url does not start with http:// or https://') })`

**Comment by JTF4** [comment]
I had a brain fart when I did the logic and used an `or` operator instead of an `and` operator. With the `or` operator it will always  report an error. I've changed it to an `and` operator to fix it.

File: lib/Service/Rest.js (hunk 1 above)
Comment (Julusian) [reply lib/Service/Rest.js]: Yeah probably should be

Comment (JTF4) [reply lib/Service/Rest.js]: I'll update that then.

Comment (krocheck) [reply lib/Service/Rest.js]: @JTF4 are we still expecting this update?

Comment (JTF4) [reply lib/Service/Rest.js]: @krocheck yes. I got bogged down with some other projects. I'll probably update the commit tomorrow or the day after.

Comment (JTF4) [reply lib/Service/Rest.js]: @krocheck would it be worth promisifying the process? Or is that I thing for a future companion release?

**Pull request closed without merging**